config/main.py

```diff
@@ -793,7 +793,7 @@ def _wait_until_clear(tables, interval=0.5, timeout=30, verbose=False):
                     click.echo("Some entries matching {} still exist: {}".format(table, keys[0]))
                 time.sleep(interval)
         empty = (non_empty_table_count == 0)
-    if not empty:
+    if not empty and timeout:
```

Comment:
Suggest checking `timeout != 0`

Reply:
I've slightly updated the logic to also return True in case timeout is 0. 

This is to be compatible with the following PR: https://github.com/sonic-net/sonic-utilities/pull/3710